src/vec.rs

```diff
@@ -143,9 +143,8 @@ impl<'data, T: Send> IndexedParallelIterator for Drain<'data, T> {
 
             // Create the producer as the exclusive "owner" of the slice.
             let producer = {
-                // Get a correct borrow lifetime, then extend it to the original length.
-                let mut slice = &mut self.vec[start..];
-                slice = slice::from_raw_parts_mut(slice.as_mut_ptr(), self.range.len());
+                let ptr = self.vec.as_mut_ptr().add(start);
+                let slice: &'data mut [T] = slice::from_raw_parts_mut(ptr, self.range.len());
```

Comment:
Doesn't this slice alias `Drain`'s `&'data mut Vec<T>` that will be used in `Drop`? The idea before was to have a properly re-borrowed (stacked?) lifetime just for the more local scope. I'm not sure whether pointers represent that nesting when you create a new slice.

Another way to do this is with a function that ties the input and output lifetimes together. (elided or explicit)

Reply:
You marked this comment about `'data` aliasing as resolved, but I don't see how it was addressed in any way...

Reply:
Same. It's been a while, I don't know why I did that.

I'm not entirely sure what you mean by `'data` aliasing. I agree that this code could become dangerous if the lifetime is wrong. But there's no aliasing issue in terms of Stacked Borrows if you get lifetimes wrong. But in any case, I think that a helper function is a good idea here, especially because I'm struggling to come up with a good name for it.

Reply:
In my commit, I moved this to a `DrainProducer::from_vec` to borrow the lifetime from `&mut Vec`.

Reply:
Oh jeez I see this slipped out of my unreads again without me responding. In any case, I like what you've done here and thanks for plowing ahead.

Reply:
No worries, thanks for checking back!